pkg/controllers/metrics/pod/controller.go

```diff
@@ -133,21 +165,65 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		},
 	})
 	c.recordPodStartupMetric(pod)
+	c.recordPodBoundMetric(pod, labels)
```

Comment:
We aren't ruling out pods that Karpenter doesn't think can schedule. As it stands right now, you're going to have metrics that are going to increase forever because Karpenter will never be able to schedule them. Can we build this so that we filter out pods that Karpenter thinks it has no chance at getting to?